starlette/testclient.py

```diff
@@ -152,11 +166,35 @@ async def _asgi_receive(self) -> Message:
     async def _asgi_send(self, message: Message) -> None:
         self._send_queue.put(message)
 
-    def _raise_on_close(self, message: Message) -> None:
+    def _handle_response(self, message: Message) -> None:
+        status_code: int = message["status"]
+        headers: typing.List[typing.Tuple[bytes, bytes]] = message["headers"]
+        body = []
+        while True:
+            message = self.receive()
+            assert message["type"] == "websocket.http.response.body"
+            body.append(message["body"])
+            if not message.get("more_body", False):
+                break
+        raise WebSocketDenialResponse(
+            status_code=status_code,
+            headers=headers,
+            content=b"".join(body),
+        )
+
+    def _raise_on_close(
+        self, message: Message, reject: typing.Optional[bool] = None
+    ) -> None:
         if message["type"] == "websocket.close":
-            raise WebSocketDisconnect(
-                message.get("code", 1000), message.get("reason", "")
-            )
+            if not reject:
+                raise WebSocketDisconnect(
+                    message.get("code", 1000), message.get("reason", "")
+                )
+            else:
+                raise WebSocketDisconnect(
+                    code=message.get("code", 1000),
+                    reason=message.get("reason"),
+                )
```

Comment:
Isn't this the same exception? The default empty string proves something?

Reply:
it is, previously we used to raise a different exception there, to indicate a special "kind" of close.  Since the distinction is now gone, I can simplify.

Reply:
Ok, I see you already did all that, so never mind.